src/filesystem_helper.cpp

```diff
@@ -474,6 +475,10 @@ bool remove_all(const path & p)
   return 0 == ret && false == file_options.fAnyOperationsAborted;
 #else
   DIR * dir = opendir(p.string().c_str());
+  if (dir == nullptr) {
```

Comment:
what happened before if `dir` was `nullptr`?

Reply:
that is a good question, above lines to check if the path is directory, and if not, it tries to remove that. because of this, i think it returns false with this check. but i though that null check should be added after `opendir` since there would be a racy condition.

Reply:
I think this would have crashed previously if `opendir` ever returned `nullptr`.  In particular, `readdir` crashes on a NULL passed-in pointer, so I think if we got e.g. `EACCES`, this would have crashed.  This is a good fix to have.